xbmc/platform/posix/filesystem/SMBDirectory.h

```diff
@@ -29,5 +29,7 @@ class CSMBDirectory : public IDirectory
 
 private:
   int OpenDir(const CURL &url, std::string& strAuth);
+  /* Should be called with smb lock held */
+  bool IsSmbValid() const { return smb.IsSmbValid(); };
```

Comment:
Are these helper methods really needed? 

You have a comment to hold the lock but when looking at this it's not obvious that you hold the lock before calling this method instead of including the lock within this method.

Reply:
> Are these helper methods really needed?
Not strictly, we can remove it for now.
Anyway, Samba interface needs to be refactored and updated, so it could be added later, when it becomes class local variable.

> You have a comment to hold the lock but when looking at this it's not obvious that you hold the lock before calling this method instead of including the lock within this method.

Lock is not really required, but it makes sense to hold it otherwise state could be changed right after getting current value.

I'll update the PR.